src/openzeppelin/token/erc721/library.cairo

```diff
@@ -70,490 +70,494 @@ end
+        end
+        let (caller) = get_caller_address()
+        let (is_approved) = _is_approved_or_owner(caller, token_id)
+        with_attr error_message("ERC721: either is not approved or the caller is the zero address"):
+            assert_not_zero(caller * is_approved)
+        end
+        # Note that if either `is_approved` or `caller` equals `0`,
+        # then this method should fail.
+        # The `caller` address and `is_approved` boolean are both field elements
+        # meaning that a*0==0 for all a in the field,
+        # therefore a*b==0 implies that at least one of a,b is zero in the field
 
-func ERC721_transferFrom{
-        pedersen_ptr: HashBuiltin*,
-        syscall_ptr: felt*,
-        range_check_ptr
-    }(from_: felt, to: felt, token_id: Uint256):
-    alloc_locals
-    with_attr error_message("ERC721: token_id is not a valid Uint256"):
-        uint256_check(token_id)
-    end
-    let (caller) = get_caller_address()
-    let (is_approved) = _is_approved_or_owner(caller, token_id)
-    with_attr error_message("ERC721: either is not approved or the caller is the zero address"):
-        assert_not_zero(caller * is_approved)
+        _safe_transfer(from_, to, token_id, data_len, data)
+        return ()
     end
-    # Note that if either `is_approved` or `caller` equals `0`,
-    # then this method should fail.
-    # The `caller` address and `is_approved` boolean are both field elements
-    # meaning that a*0==0 for all a in the field,
-    # therefore a*b==0 implies that at least one of a,b is zero in the field
-
-    _transfer(from_, to, token_id)
-    return ()
-end
 
-func ERC721_safeTransferFrom{
-        pedersen_ptr: HashBuiltin*,
-        syscall_ptr: felt*,
-        range_check_ptr
-    }(
-        from_: felt,
-        to: felt,
-        token_id: Uint256,
-        data_len: felt,
-        data: felt*
-    ):
-    alloc_locals
-    with_attr error_message("ERC721: token_id is not a valid Uint256"):
-        uint256_check(token_id)
-    end
-    let (caller) = get_caller_address()
-    let (is_approved) = _is_approved_or_owner(caller, token_id)
-    with_attr error_message("ERC721: either is not approved or the caller is the zero address"):
-        assert_not_zero(caller * is_approved)
-    end
-    # Note that if either `is_approved` or `caller` equals `0`,
-    # then this method should fail.
-    # The `caller` address and `is_approved` boolean are both field elements
-    # meaning that a*0==0 for all a in the field,
-    # therefore a*b==0 implies that at least one of a,b is zero in the field
-
-    _safe_transfer(from_, to, token_id, data_len, data)
-    return ()
-end
+    #
+    # Internals
+    #
 
-func ERC721_mint{
-        pedersen_ptr: HashBuiltin*,
-        syscall_ptr: felt*,
-        range_check_ptr
-    }(to: felt, token_id: Uint256):
-    with_attr error_message("ERC721: token_id is not a valid Uint256"):
+    func _only_token_owner{
+            pedersen_ptr: HashBuiltin*,
```

Comment:
```suggestion
    func assert_only_token_owner{
            pedersen_ptr: HashBuiltin*,
```
Since this is a guard, we should make it explicit with `assert`